File: examples/university-demo/src/main/java/org/axonframework/examples/demo/university/faculty/write/subscribestudent/SubscriptionId.java
package org.axonframework.examples.demo.university.faculty.write.subscribestudent;

import org.axonframework.examples.demo.university.shared.ids.CourseId;
import org.axonframework.examples.demo.university.shared.ids.StudentId;

record SubscriptionId(CourseId courseId, StudentId studentId) {

}

File: examples/university-demo-kotlin/src/main/kotlin/shared/ids/SubscriptionId.kt
package org.axonframework.examples.university.shared.ids

import org.axonframework.examples.university.faculty.FacultyTags
import org.axonframework.messaging.eventstreaming.Tag

data class SubscriptionId(
    val studentId: StudentId,
    val courseId: CourseId,
) {
    fun studentTag() = Tag.of(FacultyTags.STUDENT, studentId.toString())
    fun courseIdTag() = Tag.of(FacultyTags.COURSE, courseId.toString())
}


